agentlib_mpc/data_structures/casadi_utils.py

```diff
@@ -220,7 +260,21 @@ def temporary_directory(path):
         os.chdir(old_pwd)
 
 
-def compile_ipopt_solver(bat_file: Path, name: str, optimizer: ca.Function) -> str:
+def make_casadi_nlp(
+    problem: Union[dict, str],
+    solver: str,
+    opts: dict,
+    problem_type: Literal["nlp", "qp"] = "nlp",
+):
+    ca_sol = ca.nlpsol if problem_type == "nlp" else ca.qpsol
+    try:
+        solver = ca_sol("mpc", solver, problem, opts)
+    except RuntimeError:
+        solver = ca_sol("mpc", solver, problem, {**opts, "expand": False})
```

Comment:
Nice, I like the `expand=True` default. But maybe we should log a warning or something similar so users see that `expand=False`?

Reply:
going to add this